subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaIntegTest.groovy

```diff
@@ -201,6 +201,67 @@ class MavenPublishJavaIntegTest extends AbstractMavenPublishIntegTest {
         ]
     }
 
+    def "can publish java-library with dependency constraints"() {
+        given:
+        createBuildScripts("""
+
+            ${jcenterRepository()}
+
+            dependencies {
+                api "org.springframework:spring-core:2.5.6"
+                constraints {
+                    implementation "commons-logging:commons-logging:1.2"
+                }
```

Comment:
Let's add a few more dependency constraints here, to make sure they are published correctly:
- `api` vs `implementation`
- Direct dependency with no version + dependency constraint providing version. (I guess that will fail when resolving the published POM, so might need a separate test case for now).
- Constraint for `implementation` that overrides a constraint for `api` (same module, different version)
- Constraint with strict version

Reply:
I am adding these as part of #3630 to also treat the consumption in the tests.